cmd/csi-node-driver-registrar/main.go

```diff
@@ -69,6 +71,8 @@ var (
 	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including the health check indicating whether the registration socket exists, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--health-port` and `--http-endpoint` can be set.")
 	showVersion             = flag.Bool("version", false, "Show version.")
 	mode                    = flag.String("mode", ModeRegistration, `The running mode of node-driver-registrar. "registration" runs node-driver-registrar as a long running process. "kubelet-registration-probe" runs as a health check and returns a status code of 0 if the driver was registered successfully, in the probe definition make sure that the value of --kubelet-registration-path is the same as in the container.`)
+	enableProfile           = flag.Bool("enable-pprof", false, "enable pprof profiling")
+	profilePort             = flag.Int("pprof-port", 6065, "port for pprof profiling")
```

Comment:
I suggest using `http-endpoint` instead, similar to controller sidecars: https://github.com/kubernetes-csi/external-provisioner/blob/7ae61466a611061ba70da2679358c8bfc8546b6f/cmd/csi-provisioner/csi-provisioner.go#L85

Future HTTP handlers can reuse this endpoint. With that format both `enable-pprof` and `http-endpoint` need to be set explicitly.

Reply:
@verult Not sure I understand! The `http-endpoint` is used explicitly for healthcheck and can be optionally used for metrics in the `node-driver-registrar` too. `pprof-port` is explicitly for pprof. Is there a need to reuse http handlers? Let me know what you think and I can make the changes. 

https://github.com/kubernetes-csi/node-driver-registrar/blob/c63de8714b1b5df76e3310b6224c814de6e11438/cmd/csi-node-driver-registrar/main.go#L68-L69

Reply:
`http-endpoint` was intended to be the single endpoint serving all HTTP requests in each controller sidecar, because we don't want each handler to require a different port. It's may not be only for health check and metrics - it just so happens that today those are the necessary handlers, but there could be others in the future.

In the case of node-driver-registrar, following the controller sidecar pattern, `http-endpoint` would also be the single http endpoint, but the only handler it serves is pprof.

Another reason to do this, other than consistency, is to avoid having to deprecate the `pprof-port` flag when we need to add a new handler.

Hopefully it's not too complicated to set up a common mux and add the pprof handler. Let me know what you think.

Reply:
sounds good! Let me update it to use the `http-handler`.

Reply:
@verult Added a common mux. Let me know what you think.